live_site.py

```diff
@@ -0,0 +1,433 @@
+		return None
+			
+
+def check_domain(domain_name):
+
+	my_domains = get_all_domains()
+	domain_exists = ''
+
+	# Checks if the domain name already exists or not
+	for domain, details in enumerate(my_domains['domains']):
+		for key, value in details.items():
+			x = details['name']            
+			if domain_name != x: continue
+			domain_exists = x
+			return domain_exists
+		else:
+			return None	
+	else:
+		return None
+
+
+def will_add_domain(droplet, domain_name):
+		# IP address is needed to create the domain name, so we call the function get_hostname_ip()
+		ip = get_hostname_ip(droplet)
+		# Just want to make sure the ip_address is a string, so I convert it anyway.
+		ip_address = str(ip)  
+
+		# This adds/creates the domain name
+		new_domain = add_new_domain(domain_name, ip_address)
+
+		if new_domain is not None:
+			click.secho('>>> Your new domain name {} has been created with IP Address {}. '.format(domain_name, ip_address), fg='bright_green')
+		else:
+			click.secho('[!] Request Failed', fg='bright_red')
+
+
+def create_domain(droplet, domain_name):
+
+	domain_exists = check_domain(domain_name)
+	click.echo(domain_exists)
+
+	if domain_exists == domain_name:
+		click.secho('>>> Your domain name {} already exists!'.format(domain_name), fg='bright_yellow')
+		click.secho('>>> We are now removing it to droplet IP mapping ...', fg='bright_yellow')
+		del_domain(domain_name)
+		click.secho('Creating a new domain name now...', fg='bright_yellow')
+		will_add_domain(droplet, domain_name)
+	else:
+		click.echo('...')
+		will_add_domain(droplet, domain_name)
+
+
+def add_new_domain_record(type, name, data, ttl, domain_name):
+
+	api_url = '{0}domains/{1}/records'.format(api_url_base,domain_name)    
+
+	records = {"type":type,"name":name,"data":data,"ttl":ttl}
+
+	response = requests.post(api_url, headers=headers, json=records) 
+
+	if response.status_code == 201:
+		records = json.loads(response.content)
+		return records
+	else:
+		click.secho('[?] Unexpected Error: [HTTP {0}]: Content: {1}'.format(response.status_code, response.content), fg='bright_red')
+		return None                     
+		
+
+def add_domain_rec(droplet, domain_name):
+	# Get the IP Address associated with jediluke.space
+	ip = get_hostname_ip(droplet)
+	rec_data = str(ip)
+
+	add_response = add_new_domain_record('A','www',rec_data,1800, domain_name)
+
+	if add_response is not None:
+		click.secho('>>> Your A Record has been created... ', fg='bright_green')
+		for k, v in add_response.items():
+			click.secho('  {0}:{1}'.format(k, v), fg='bright_green')
+	else:
+		click.secho('[!] Request Failed', fg='bright_red')
+
+def set_nginx(droplet, domain_name):
+
+	host = get_hostname_ip(droplet)
+
+	client = SSHClient(host, user='root', password='', pkey=ssh_key,port=22)
+	click.secho('Connecting to your server...', fg='bright_yellow')
+
+	def upload():
```

Comment:
I think this might be one of your problems - Python expects us to use tabs, new lines and spaces (collectively called whitespace) to lay out our functions etc. If you have accidentally tabbed this in, it might be giving you problems.

It also might not, I don't have a python shell open - but even if it compiles, just put all the functions up against the hard left side of the file, it makes it easier to read

Reply:
If you want to use this function here, then just move the whole upload function outside this block, and call it from within it